tools/dist/promote/_promote.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+if [ -z ${srcdir+x} ]; then
+  echo "\$srcdir is not set"
+  exit 1
+fi
+
+if [ -z ${dstdir+x} ]; then
+  echo "\$dstdir is not set"
+  exit 1
+fi
+
+version=$1
+
+for subdir in $(cd $srcdir && ls); do
+
+  if [ -d "${srcdir}/${subdir}" ]; then
+    resha=no
+
+    if [ "X${version}" != "X" ] && [ "X${version}" != "X${subdir}" ]; then
+      continue
+    fi
+
+    for donefile in $(cd ${srcdir}/${subdir} && ls *.done 2> /dev/null); do
+      doneref=$(echo $donefile | sed 's/\.done$//')
+
+      if [[ -f "${srcdir}/${subdir}/${doneref}" || -d "${srcdir}/${subdir}/${doneref}" ]]; then
+        echo "Promoting ${subdir}/${doneref}..."
+        mkdir -p "${dstdir}/${subdir}"
+        rm -rf "${dstdir}/${subdir}/${doneref}"
+        cp -dr "${srcdir}/${subdir}/${doneref}" "${dstdir}/${subdir}/${doneref}"
```

Comment:
I'm sure there was a good reason I used `-dr` here, I'd normally reach for `-a` in almost any situation and am not sure why I'm not using it here .. I'm a tiny bit afraid to find out what I was thinking!